pandas/tests/indexes/interval/test_interval.py

```diff
@@ -482,11 +493,18 @@ def test_slice_locs_decreasing_int64(self):
     def test_slice_locs_decreasing_float64(self):
         self.slice_locs_cases([(2., 4.), (1., 3.), (0., 2.)])
 
-    # To be removed, replaced by test_interval_new.py (see #16316, #16386)
-    def test_slice_locs_fails(self):
-        index = IntervalIndex.from_tuples([(1, 2), (0, 1), (2, 3)])
+    @pytest.mark.parametrize('query', [
+        [0, 1], [0, 2], [0, 3], [3, 1], [3, 4], [0, 4]], ids=str)
+    @pytest.mark.parametrize('tuples', [
+        [(0, 2), (1, 3), (2, 4)],  # increasing overlapping
+        [(2, 4), (1, 3), (0, 2)],  # decreasing overlapping
+        [(0, 2), (0, 2), (2, 4)],  # sorted duplicates
+        [(0, 2), (2, 4), (0, 2)],  # unsorted duplicates
+        [(0, 2), (0, 2), (2, 4), (1, 3)]])  # another unsorted duplicates
```

Comment:
Slight modification of the deleted test to parametrize over tuples.  Makes the test code itself cleaner.